### PR TITLE
JS3 Week 3 / Yalcin TOK

### PR DESCRIPTION
**Pull request by yalcintok**

### CODE REVIEW

File: Week3/part 1/updateWeek2Hw.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
+   'use strict';
+
+{
+  async function fetchJSON(url, cb) {
```
Comment (ArcoMul) [inline Week3/part 1/updateWeek2Hw.js]: Some things are not going right in the function:
- The await on axios is outside of the try/catch block. If axios throws an error the exception will not be caught by the try/catch in this function
- You are now reading the status value of the response, but actually (when the try/catch is around the axios await) axios will throw an error when the status code is not in the 2XX range (so it would be 'catched' by the try/catch around axios)

So to summarize: it's better to move axios into the try catch, then if it goes fine just return response.data, but when axios encounters an error it will enter the catch part of your try/catch and the error can be handled nicely.

File: Week3/part 1/updateWeek2Hw.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    const ul = createAndAppend('ul', parent)
+    const repoEls = createAndAppend('li', ul, { class: 'list-items' });
+    addRepoInformation('Repository:', repo.name, repoEls, repo.html_url);
+    addRepoInformation('Description:', repo.description, repoEls);
+    addRepoInformation('Forks:', repo.forks, repoEls);
+    addRepoInformation('Updated:', arrangeDate(repo.updated_at), repoEls);
+  }
+
+  function appendContributor() {
+    fetchJSON(currentRepo.contributors_url)
```
Comment (ArcoMul) [inline Week3/part 1/updateWeek2Hw.js]: Async/await can be used here

File: Week3/part 1/updateWeek2Hw.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        data.forEach(contr => addRepoContributors(contr, contributorField));
+      })
+      .catch(err => {
+        console.log(err);
+          return;
+      })
+  };
+
+  function main(url) {
+    fetchJSON(url)
```
Comment (ArcoMul) [inline Week3/part 1/updateWeek2Hw.js]: Async/await can be used here as well

File: Week3/part 1/updateWeek2Hw.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    fetchJSON(url) 
+      .then(repos => {
+        
+        let head = document.querySelector('.head');
+        head.innerHTML = 'HYF Repositories'
+        const select = createAndAppend('select', head);
+        receivedRepos = repos.sort((a, b) => a.name.localeCompare(b.name));
+        receivedRepos.forEach(repo => renderRepoDetails(repo, repoField));
+        select.addEventListener('change', function(event) {
+          console.log(event.target.value);
```
Comment (ArcoMul) [inline Week3/part 1/updateWeek2Hw.js]: Please remove console.logs before creating pull requests :-)

File: Week3/part 1/updateWeek2Hw.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+          for (let i = 0; i < receivedRepos.length; i++) {
+            if (event.target.value == receivedRepos[i].id) {
```
Comment (ArcoMul) [inline Week3/part 1/updateWeek2Hw.js]: I think you already know better array methods to find a specific item. Look over the ones there are in the menu on the left: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array

File: Week3/part 1/updateWeek2Hw.js (hunk 1 above)
Comment (yalcintok) [reply Week3/part 1/updateWeek2Hw.js]: Thank you @ArcoMul. I got it what you recommended me. And all mistakes were fixed. I think now is beter than last.

File: Week3/part 1/updateWeek2Hw.js (hunk 2 above)
Comment (yalcintok) [reply Week3/part 1/updateWeek2Hw.js]: Yeap I appended.

File: Week3/part 1/updateWeek2Hw.js (hunk 3 above)
Comment (yalcintok) [reply Week3/part 1/updateWeek2Hw.js]: fixed.

File: Week3/part 1/updateWeek2Hw.js (hunk 4 above)
Comment (yalcintok) [reply Week3/part 1/updateWeek2Hw.js]: Fixed also :)

File: Week3/part 1/updateWeek2Hw.js (hunk 5 above)
Comment (yalcintok) [reply Week3/part 1/updateWeek2Hw.js]: I analyzed that.  Yes right we know almost all of them but exactly have to improve day to day.

**Comment by wouterkleijn** [comment]
Thanks for the updates. All good @ArcoMul ?

**Comment by ArcoMul** [comment]
Thanks for the fixes @yalcintok 👍
